### PR TITLE
Add initial AppVeyor support.

### PR DESCRIPTION
**Pull request by XhmikosR**
Things to do:

- [x] Add AppVeyor to the repo and then I'll change the badge link
- [x] **Make sure you check the option to skip builds if `appveyor.yml` is missing in AppVeyor project settings**
- [x] Should we use `.appveyor.yml` or the default `appveyor.yml`?
- [x] Should we run all tests? Currently I only run `make test-ci`
- [x] Which Node.js versions should we test?
- [x] Which platforms? x86/x64/both? I went with x64 for now.
- [ ] Notifications? Same as Travis? Someone should take care of this themselves since we can't disable notifications via the config file.
- [ ] Fix tests to be cross-platform

/CC @xtuc @hzoo

### CODE REVIEW

**Comment by xtuc** [comment]
I'm not sure `Make` is available by default, I used to install it: https://github.com/xtuc/webassemblyjs/blob/master/appveyor.yml#L5

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - ps: Install-Product node $env:nodejs_version
+  - yarn install --ignore-engines
+
+build: off
+
+test_script:
+  - set PATH=C:\msys64\usr\bin;%PATH%
+  - node -v
+  - yarn -v
+  - make lint
```
Comment (xtuc) [inline appveyor.yml]: I would remove linting, the platform has no impact

File: README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 <p align="center">
   <a href="https://travis-ci.org/babel/babel"><img alt="Travis Status" src="https://img.shields.io/travis/babel/babel/master.svg?label=travis&maxAge=43200"></a>
+  <a href="https://ci.appveyor.com/project/XhmikosR/babel"><img alt="AppVeyor Status" src="https://img.shields.io/appveyor/ci/XhmikosR/babel/master.svg?label=AppVeyor%20build&maxAge=43200"></a>
```
Comment (xtuc) [inline README.md]: XhmikosR/babel -> babel/babel

**Comment by XhmikosR** [comment]
@xtuc: if make wasn't available, we would get an error ;) See the `PATH` setup; I'm using MSYS2 folder.

File: README.md (hunk 2 above)
Comment (XhmikosR) [reply README.md]: Please see the todo in my OP.

**Comment by xtuc** [comment]
Yes that's right, I haven't realized that you already tested it :+1:

File: appveyor.yml (hunk 1 above)
Comment (XhmikosR) [reply appveyor.yml]: ok

**Comment by XhmikosR** [comment]
For reference, #8184 is a Windows only issue which is clearly shown in the test failures on AppVeyor, so it could have been caught. :)

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+version: "{build}"
+
+clone_depth: 10
+
+environment:
+  PLATFORM: "x64"
+  matrix:
+    - NODEJS_VERSION: "10"
+    - NODEJS_VERSION: "8"
+    - NODEJS_VERSION: "6"
```
Comment (hzoo) [inline appveyor.yml]: Ok with only testing x64 or (node v6/latest node) since we just want to test windows compat vs. diffferences in node

**Comment by XhmikosR** [comment]
Even with #8281 merged to my branch, there still RegExp failures on Windows apart from the paths mismatches/line endings failures.

**Comment by gengjiawen** [comment]
@XhmikosR Can you rebase this pr to trigger ci again ?

**Comment by XhmikosR** [comment]
It's already rebased against the latest upstream master so I doubt anything will change, but anyway.

**Comment by XhmikosR** [comment]
Most failures are related to this function in babel-core/test

```js
function fixture(...args) {
  return path.join(__dirname, "fixtures", "config", ...args);
}
```

And the rest are just path differences like

```diff
    - var _jsxFileName = "/fake/path/mock.js";
    + var _jsxFileName = "C:\\fake\\path\\mock.js";
```

**Comment by gengjiawen** [comment]
I am confused, the related pr has been merged. Is there another issue on windows ?
cc @hzoo @xtuc @loganfsmyth

**Comment by XhmikosR** [comment]
Yes, there are differences in RegExp tests and paths in general. Just check the build output.

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+version: "{build}"
+
+clone_depth: 10
+
+environment:
+  PLATFORM: "x64"
+  matrix:
+    # check the current Node.js release and the minimum supported one
+    - NODEJS_VERSION: "10"
+    - NODEJS_VERSION: "6"
```
Comment (gengjiawen) [inline appveyor.yml]: Why not add node version 8 ?

Comment (XhmikosR) [reply appveyor.yml]: Because @hzoo didn't want to add it.

Comment (gengjiawen) [reply appveyor.yml]: Any special reason, travis ci has node 8. And node 8 is lts release.

**Comment by XhmikosR** [comment]
@hzoo @loganfsmyth: please lock the PR until we get it merged.

File: appveyor.yml (hunk 3 above)
Comment (gengjiawen) [reply appveyor.yml]: Node 8 is LTS release too, why not ? We have more confidence adding node 8 other than remove it.

**Comment by XhmikosR** [comment]
@hzoo: can you enable cache in PRs in AppVeyor settings? This won't have any effect until the builds pass, but still.

**Pull request closed without merging**